### PR TITLE
UserProfileDetails adding and saving loggic added.

### PR DESCRIPTION
**Pull request by Washubrnfg85**
org.linksharing.server.users.UserProfileDetailsRepository created to maintain UserProfileDetailes DB job.
AuthController -> UserProfileDetails object generates and saves in parallel with User. Same email used as Id for both entities.
ApiRestController -> data from post request updates UserProfileDetails and saves it to DB.

### CODE REVIEW

File: src/main/java/org/linksharing/server/auth/AuthController.java [hunk 1]
```diff
@@ -47,7 +51,12 @@ public String registerUser(@Valid User user, BindingResult result) {
             return "redirect:/login?userExists";
         }
 
+        UserProfileDetails userProfile = new UserProfileDetails();
+        userProfile.setEmail(user.getEmail());
+        userProfile.setFirstName(user.getUsername());
+
         userRepository.save(user);
+        profileRepository.save(userProfile);
```
Comment (eugenesmkn) [inline src/main/java/org/linksharing/server/auth/AuthController.java:58]: Don't have to do that here. Move this to `getProfileDetails`.

File: src/main/java/org/linksharing/server/http/ApiRestController.java [hunk 2]
```diff
@@ -39,7 +45,16 @@ ResponseEntity<?> getProfileDetails(Principal user) {
...
+
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+
+        userProfile.setFirstName(profileDetails.getFirstName());
+        userProfile.setLastName(profileDetails.getLastName());
+        userProfile.setPublicEmail(profileDetails.getPublicEmail());
+
+        profileRepository.save(userProfile);
+
         return null;
```
Comment (eugenesmkn) [inline src/main/java/org/linksharing/server/http/ApiRestController.java]: Should return user profile details data here (ResponseEntity).
The same is for `getProfileDetails`.
This data should be in html inputs, no additional divs are required.

**Review by eugenesimakin — CHANGES_REQUESTED** [review]

File: src/main/java/org/linksharing/server/auth/AuthController.java (hunk 1 above)
Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/auth/AuthController.java:58]: Why shouldn't we? It seems logical to me. When new user registers, his profile creates and goes to DB at the same time. 
No additional checks needed later during CRUD operations. If we get User by Id, we 100% sure that he already has Profile.

File: src/main/java/org/linksharing/server/http/ApiRestController.java (hunk 2 above)
Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/http/ApiRestController.java]: This one done. That was easier than I thought... Don't know how vue handles it, but looks like AI built in it)))

File: src/main/java/org/linksharing/server/auth/AuthController.java (hunk 1 above)
Comment (eugenesmkn) [reply src/main/java/org/linksharing/server/auth/AuthController.java:58]: This doesn't look to me as necessary thing to do but sure, let's keep it that way.

File: src/main/java/org/linksharing/server/auth/AuthController.java [hunk 3]
```diff
@@ -47,7 +51,12 @@ public String registerUser(@Valid User user, BindingResult result) {
             return "redirect:/login?userExists";
         }
 
+        UserProfileDetails userProfile = new UserProfileDetails();
+        userProfile.setEmail(user.getEmail());
+        userProfile.setFirstName(user.getUsername());
```
Comment (eugenesmkn) [inline src/main/java/org/linksharing/server/auth/AuthController.java]: Let's use the username for user's public url (the url when a user sees the profile details and links, the url to share). For example, http://localhost:8080/u/username.

So, first name is not username.

**Review by eugenesimakin — CHANGES_REQUESTED** [review]

File: src/main/java/org/linksharing/server/auth/AuthController.java (hunk 3 above)
Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/auth/AuthController.java]: Ok, I'll save it for tomorrow

Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/auth/AuthController.java]: Don't quite understand logic. The URL   http://localhost:8080/u/username   should be unique for each user. So username should be unique for each user also, as it's part of URL. But it's not, cause we didn't provide that.
Or maybe I don't get? What is /u/ in URL ?

Comment (eugenesmkn) [reply src/main/java/org/linksharing/server/auth/AuthController.java]: Yes, you're right. See #12

**Review by eugenesimakin — APPROVED** [review]

**Pull request merged**